README.md

```diff
@@ -74,39 +44,176 @@ This program was created with the following tech stack and packages:
+
+The Speaking Portal API generates an MP4 animation from the speech file and text file received from Kukarella's
+TTS process. Text and audio inputs are first processed and converted into a json file containing phonemes and their
+timings. This processing is done via the [Rhubarb library](https://github.com/DanielSWolf/rhubarb-lip-sync). (Phonemes
+are the smallest unit of sounds that uniquely identify distinct sounds within words.) Once a phoneme file is created,
+the animation process then identifies which avatar assets to use based on the phoneme. Other features, such as blinks
+and pose changes are added, and then all image assets are rendered in a video file through ffmepg.
+
+### API
+
+When the API receives a request, the uploaded files are stored in the `/tmp` directory. The API code then reads these inputs and converts them to the appropriate format for Rhubarb and ffmpeg. Upon conversion, the animation process is started by sending all user inputs to the `main.ts` file. 
+
+Here's a typical example of the JSON sent in a POST request.
+
+```markdown
+{
+  fieldname: 'audio',
+  originalname: 'en-Amber.wav',
+  encoding: '7bit',
+  mimetype: 'audio/wave',
+  destination: './tmp',
+  filename: 'f8fc3bc5c3f0fbc70ab0899d0dc0c860',
+  path: 'tmp\\f8fc3bc5c3f0fbc70ab0899d0dc0c860',
+  size: 9049278
+} {
+  fieldname: 'text',
+  originalname: 'en-text.txt',
+  encoding: '7bit',
+  mimetype: 'text/plain',
+  destination: './tmp',
+  filename: 'e05b5794e54ca3810b071b87bc1449e1',
+  path: 'tmp\\e05b5794e54ca3810b071b87bc1449e1',
+  size: 1363
+} phonetic
+```
+
+#### POST Request Properties
+
+The SPP API is the first step in the animation process. Kukurella requests are sent to the SPP API as a POST with the
+following required properties:
+
+| Field      | Type | Description     |
+| :---:       |    :----:  |        :---   |
+| `audio`      | Wav file       | A file containing audio of the voice the user wishes to use for the generated mp4 output |
+| `text`   | Txt file       | A file containing a textual script of what is being communicated in the Wav file      |
+| `recognizer`   | String        | The recognizer for the language processor. Options include default as `english` and non-default `phonetic`. The English-only recognizer is slower but more accurate. The phonetic recognizer is faster but not as accurate, and it supports over 200 languages.      |
+| `avatar` |  String  | Specifies  avatar animation choice for video output. Options include default `barb` or non-default `boy` |
+
+#### API Constraints
+
+- This API only responds to POST calls
+- This process is computationally intensive, as such expect a wait time etween requests. More user requests will result
+in slower processing times
+- The script text file should match the given audio file as closely as possible in terms of content. Failure to do so
+could lead to errors.
+- Port 3000 must be referenced in addition to the IP address.
```

Comment:
I mean technically someone could update this to use whatever port they want?

Reply:
There are specific ports used for specific reasons. 3000 is technically arbitrary, but it's also known for not being used by specific things, so that's why it's typically used. If someone was told to use an arbitrary port and didn't know, then they might choose a port that was for something else and have issues. We could "recommend" instead of saying it's a "must"

Reply:
I like the recommendation rather than obligation for this :)